src/system.jl

```diff
@@ -329,6 +330,142 @@ function get_datetimes(system::System)
     return axiskeys(system.generator_time_series.offer_curve, 2)
 end
 
+get_zones(system::System) = system.zones
+
+"Returns a `Dictionary` with zonal regulation requirements indexed by zone number."
+function get_regulation_requirements(system::System)
+    return map(system.zones) do zone
+        zone.reg
+    end
+end
+
+"Returns a `Dictionary` with zonal operating reserve requirements indexed by zone number."
+function get_operating_reserve_requirements(system::System)
+    return map(system.zones) do zone
+        sum([zone.reg, zone.spin])
```

Comment:
I don't know how this is calculated.  Is it a function of the information contained in the `Zone` type?

Reply:
Oh yeah, I mixed things, there's another requirement that is just over reg + spin, but operating reserve is over reg + spin + supp (and the on vs off distinction doesn't make sense in this case, the docstring there is a bit confusing, since a generator cannot simultaneously provide online and offline supplemental reserve).

The problem here is that the `Zone` type is not defined how it should, I let this slip in a previous PR. There are no individual requirements for each service type in MISO, what they have is
1. regulation requirement (just reg)
2. operating reserve requirement (reg + spin + supp)
3. good utility practice requirement (reg + spin)

so we need to redefine `Zone` to have these three requirements

Reply:
Just to check my understanding in code.  The zone should have just three requirements
```julia
struct Zone
    number::Int64
    regulation::Float64
    operating_reserve::Float64
    good_utility::Float64 # we don't model this at the moment right?
end
```
And then the `get_operating_reserve_requirements` function would just be:
```julia
function get_operating_reserve_requirements(system::System)
    return map(system.zones) do zone
        zone.operating_reserve
    end
end
```
And when we come to model the operating reserve requirement that would involve getting time series data from all four of the ancillary services time series (`asm_regulation`, `asm_spinning`, `asm_sup_on` and `asm_sup_off`).  Does this seem right?

Reply:
Yes, that makes sense to me.

># we don't model this at the moment right?

It seems we don't, and I'm not sure why. We do define these requirements on the data side [here](https://gitlab.invenia.ca/invenia/research/FullNetworkDataPrep.jl/-/blob/master/src/build/reserve_requirements.jl#L29) with a confusing name, but not on the modelling side. I'll open an issue